src/main/java/com/dev/cinema/service/mappers/ShoppingCartMapper.java

```diff
@@ -0,0 +1,45 @@
+package com.dev.cinema.service.mappers;
+
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+import com.dev.cinema.model.dto.ShoppingCartResponseDto;
+import com.dev.cinema.model.dto.TicketResponseDto;
+import com.dev.cinema.service.ShoppingCartService;
+import com.dev.cinema.service.UserService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoppingCartMapper {
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final TicketMapper ticketMapper;
+
+    @Autowired
+    public ShoppingCartMapper(UserService userService,
+                              ShoppingCartService shoppingCartService,
+                              TicketMapper ticketMapper) {
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.ticketMapper = ticketMapper;
+    }
+
+    public ShoppingCartResponseDto toShoppingCartResponseDto(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(userService.findById(userId));
```

Comment:
let's dot this in controller and pass the ShoppingCart to mapper instead of userId